pkg/rc/farm.go

```diff
@@ -108,37 +105,32 @@ func (rcm *Farm) Start(quit <-chan struct{}) {
 			// track which children were found in the returned set
 			foundChildren := make(map[fields.ID]struct{})
 			for _, rcField := range rcFields {
+				rcLogger := rcm.logger.SubLogger(logrus.Fields{
+					"rc_id": rcField.ID,
+				})
 				if _, ok := rcm.children[rcField.ID]; ok {
 					// this one is already ours, skip
-					rcm.logger.WithField("rc_id", rcField.ID).Debugln("Got replication controller already owned by self")
+					rcLogger.NoFields().Debugln("Got replication controller already owned by self")
 					foundChildren[rcField.ID] = struct{}{}
 					continue
 				}
 
-				success, err := rcm.rcStore.LockRead(rcField.ID, session)
-				if err != nil {
-					// log and break out of the loop - we can't continue if
-					// we had a real failure, our session probably disappeared
-					rcm.logger.WithErrorAndFields(err, logrus.Fields{
-						"rc_id":   rcField.ID,
-						"session": session,
-					}).Errorln("Got error while locking replication controller - session may be expired")
-					break // TODO: should we be pruning after this?
-				} else if !success {
+				err := rcm.lock.Lock(kp.LockPath(kp.RCPath(rcField.ID.String())))
```

Comment:
yeah the M is a leftover from when it was called RCManager, I'll add a commit that changes the names